src/ast/location.cpp

```diff
@@ -89,6 +89,9 @@ Location operator+(const Location &orig, const Location &expansion)
   if (expansion == nullptr) {
     return orig;
   }
+  if (orig == nullptr) {
+    return expansion;
+  }
```

Comment:
Wait, how can these references be null?

Reply:
It's actually a null `std::shared_ptr`, so perhaps a bit confusing. Like if you did:

```
auto new_loc = foo.loc + Location()
```

In this case, `expansion == nullptr`.

I think I can drop this case actually, but I should probably just keep it and add a comment that explains it regardless.

Reply:
ahh, shared_ptr, got it